src/System.Text.Json/tests/JsonPropertyTests.cs

```diff
@@ -0,0 +1,289 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Collections;
+using System.Globalization;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System.Collections.Generic;
+using System.IO;
+using Xunit;
+using System.Buffers.Text;
+using System.IO.Tests;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+
+namespace System.Text.Json.Tests
+{
+    public static class JsonPropertyTests
+    {
+        private const string CompiledNewline = @"
+";
+
+        private static readonly JsonDocumentOptions s_options =
+            new JsonDocumentOptions
+            {
+                CommentHandling = JsonCommentHandling.Skip,
+            };
+
+        private static readonly bool s_replaceNewlines =
+            !StringComparer.Ordinal.Equals(CompiledNewline, Environment.NewLine);
+
+        [Fact]
+        public static void CheckUseAfterDispose()
+        {
+            using (JsonDocument doc = JsonDocument.Parse("{\"First\":1}", default))
```

Comment:
Again, really this should reuse the tests; but then provide whatever extra tests we need that would be unique to JsonProperty.

I'd rather see these tests be white-box and minimal for the 3.0 release than see all of this code/logic duplicated.

Reply:
> 
> 
> Again, really this should reuse the tests; but then provide whatever extra tests we need that would be unique to JsonProperty.
> 

Same here, I will move assertion in existing test methods itself. Not sure of unique test cases for JsonProperty but I will think about it.